### DISCUSSION TITLE
How do I preserve innerhtml when using directives?

### DISCUSSION THREAD

**Discussion by SephVelut** [alpinejs/alpine#2964] [1. Help] [answered]
Given
```html
<p x-text="some text">
    <sub>some subscript</sub>
<p>
```

Expect rendered `some text some subscript` with html tags preserved `<p>some text <sub>some subscript</sub></p>`.

`x-text` will override all inner content. I can access the inner html using `$el.innerHTML` in the directive, but unsure where to go from them to include "some text" + innerHTML. I could do this `x-html="'some text' + $el.innerHTML"` but is there a better way?

**Comment by SimoTod** [comment, marked as answer]
You can define your own custom directive to make it less verbose but if it's just in one place, your solution is probably the best
